replica/distributed_loader.cc

```diff
@@ -580,7 +573,7 @@ class table_populator {
         }
 
         co_await smp::invoke_on_all([this] {
-            _global_table->update_sstables_known_generation(_highest_generation);
+            _global_table->update_sstables_known_generation(_highest_generation.value_or(sstables::generation_type(0)));
```

Comment:
Why 0? If the generation is to be set on table's sstables, shouldn't this be a "valid" generation, say (according to previous patch) the `new_generation(std::nullopt)`?

Reply:
hmm, or just pass `_highest_generation` itself (which contains an optional)
and let `update_sstables_known_generation` decide what to do about it.

Reply:
passing optional generation in next version